src/components/DatePicker/RangePicker.jsx

```diff
@@ -0,0 +1,317 @@
+import React, { useLayoutEffect, useRef, useMemo, useState } from 'react'
+import PropTypes from 'prop-types'
+import moment from 'moment'
+import { Flex, Typography, Icon, Box, Button, styled } from '@etvas/etvaskit'
+import { css } from 'styled-components'
+
+import { Calendar } from './Calendar'
+
+import { COMMON_FORMAT, CURRENT_MONTH_FORMAT } from './constants'
+
+export const RangePicker = ({
+  startOfTime,
+  endOfTime,
+  value,
+  displayFormat,
+  yearDisplayStart,
+  yearDisplayEnd,
+  navigationByYear,
+  disabled,
+  onChange,
+  label,
+  placeholder,
+  ...props
+}) => {
+  const wrapRef = useRef()
+  const [isExpanded, setExpanded] = useState(false)
+  const [isSettingEnd, setSettingEnd] = useState(false)
+  const [currentHover, setCurrentHover] = useState(null)
+
+  const [currentMonth, setCurrentMonth] = useState(() => {
+    const { start } = value || {}
+    const currentMonth = moment(start || moment().startOf('month'))
+    return currentMonth
+  })
+
+  const resetDateRange = val => {
+    const { start, end } = val || {}
+
+    const mStart = moment(start || moment())
+    const mEnd = moment(end || moment())
+
+    return { mStart, mEnd }
+  }
+
+  const [{ mStart, mEnd }, setDateRange] = useState(() => resetDateRange(value))
+
+  useLayoutEffect(() => {
+    const listener = event => {
+      if (
+        wrapRef?.current?.contains(event.target) ||
+        wrapRef?.current?.contains(event.srcElement)
+      ) {
+        return
+      }
+      setExpanded(false)
+      // setIsEndDate(false)
```

Comment:
is this a leftover?

Reply:
a7ace59e3bccac05c403bfc3cdebdb5db13c3147